docs/coap-demo.md

```diff
@@ -0,0 +1,165 @@
+                      }
+                      _ => {
+                          response.set_status(Status::NotFound);
+                          response.message.payload = b"Not found".to_vec();
+                      }
+                  }
+
+                  Some(response)
+              })
+              .await
+              .unwrap();
+      });
+  }
+  ```
+
+4. Run the server
+
+  ```
+  cargo run
+  ```
+
+It's not necessary, but you can check if the server is correctly up and running creating a separate Rust project for a CoAP client with the following `main.rs`:
+
+```rs
+use coap::CoAPClient;
+
+fn main() {
+    let url = "coap://192.168.1.126:5683/sensors/temp"; // Properly set your machine's IP
+    let response = CoAPClient::get(url).unwrap();
+    println!("Server reply: {}", String::from_utf8(response.message.payload).unwrap());
+}
+```
+
+## Running Akri
+
+Now it is time to install the Akri using Helm. When installing Akri, we can specify that we want to deploy the CoAP Discovery Handlers by setting the helm value `coap.discovery.enabled=true`. We also specify that we want to create a CoAP Configuration with `--set coap.configuration.enabled=true`.
+
+In the Configuration, we need to specify our CoAP server's static IP address. Otherwise, the Discovery Handler's default behaviour is to use the broadcast method: `--set coap.configuration.discoveryDetails.staticIpAddresses[0] = "192.168.1.126"`.
+
+The final CLI commands should look like this:
+
+  ```sh
+helm repo add akri-helm-charts https://deislabs.github.io/akri/
+helm install akri akri-helm-charts/akri \
+    --set coap.discovery.enabled=true \
+    --set coap.configuration.enabled=true \
+    --set coap.configuration.discoveryDetails.staticIpAddresses[0] = "192.168.1.126"
+```
+
+The Akri Discovery Handler will discover the CoAP server and create an instance. 
+
+
+```sh
+kubectl get pods
+```
+
+The result should look like this:
+
+```
+NAME                                          READY   STATUS    RESTARTS   AGE
+akri-agent-daemonset-p8w8z                    1/1     Running   0          75s
+akri-coap-discovery-daemonset-sq6pt           1/1     Running   0          75s
+akri-controller-deployment-5bc76f77d8-9ldws   1/1     Running   0          75s
+kind-control-plane-akri-coap-920f97-pod       1/1     Running   0          64s
+```
+
+To inspect more of the elements of Akri:
+
+- Run `kubectl get crd`, and you should see the CRDs listed.
+- Run `kubectl get akric`, and you should see `akri-coap`. 
+- The instances can be seen by running `kubectl get akrii` and
+  further inspected by running `kubectl get akrii akri-coap-<ID> -o yaml`
+
+## Deploy a Kubernetes application which requests the temperature via HTTP
+
+The Akri CoAP built-in Broker Pod is automatically deployed when an instance is created. The Broker exposes a RESTful endpoint which translates HTTP request to CoAP requests. To inspect the associated Kubernetes Service, run `kubectl get crd`.
```

Comment:
I added a comment to `Values.yaml` about this, but the sample brokers should not be defaults rather than specified in the demos. Can you add a `--set` line to the installation to specify the broker?